continuousflex/protocols/protocol_import_subtomograms.py

```diff
@@ -0,0 +1,277 @@
+# **************************************************************************
+# * Authors:   Rémi Vuillemot remi.vuillemot@upmc.fr
+# *
+# * This program is free software; you can redistribute it and/or modify
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 2 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+
+from pwem.protocols import ProtImportFiles
+import xmipp3.convert
+import pyworkflow
+import pwem.emlib.metadata as md
+import pyworkflow.protocol.params as params
+import pyworkflow.utils as pwutils
+import pwem.objects as emobj
+from pwem import emlib
+from os.path import exists, basename, abspath, relpath, join, splitext
+from xmipp3.convert import writeSetOfVolumes, readSetOfVolumes
+from .convert import eulerAngles2matrix, matrix2eulerAngles
+import numpy as np
+from pyworkflow.utils.path import makePath, copyFile
+from os.path import basename
+from pwem.utils import runProgram
+import json
+from ast import literal_eval as make_tuple
+import os
+
+class FlexProtImportSubtomogram(ProtImportFiles):
+    """ Protocol for importing subtomograms"""
```

Comment:
There is a problem with this protocol. Importing subtomograms already exists in scipion. If alignment information is all we are importing, then let's change the name to import alignment information. I also think such a protocol should be somewhere part of scipion and not continuousflex, since no one will be able to maintain it if it remains here.

Reply:
There is no such protocol in Scipion, not that I know. There is one in relion plugin but it takes only relion files. 
The thing is it is easier to maintain to have a single plugin where we import the angles etc. 
Because right now, we are importing these angles in different places. Following what is done for 2D particles, you import the angles together with the particles so that it is linked to a particle and you always have it when needed. 
But I agree it should be in Scipion, probably they will do that in Scipion, so maybe we can keep it here in the meantime?

Reply:
There is a protocol called import subtomograms in scipion-em-tomo. But indeed, it doesn't import the alignment information. I don't want to complicate things for you now, but it may be better if you contact Pablo and the Scipion guys and tell them about this issue and maybe someone from them will do it, and we can deprecate this protocol once its done.